### PR TITLE
Dynamic scroll to element (offset set according to notifications’ hei…

### PR DESCRIPTION
**Pull request by sczerwinski-wikia**
…ght)

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java [hunk 1]
```diff
@@ -85,21 +83,22 @@ public void scrollToBottom() {
...
+    WikiBasePageObject wikiPage = new WikiBasePageObject();
+    if (wikiPage.isBannerNotificationContainerPresent()){
+      int notificationsHeight = wikiPage.getBannerNotificationsHeight();
+      offset += notificationsHeight;
+    }
+
     try {
       js.executeScript(
-          "var x = $(arguments[0]);" + "window.scroll(0,parseInt(x.offset().top - 60));", element);
+          "var x = $(arguments[0]);" + "window.scroll(0,parseInt(x.offset().top - " + String.valueOf(offset) + "));", element);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java]: MINOR Directly append the argument of String.valueOf(). [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1153)

File: src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java [hunk 2]
```diff
@@ -85,21 +83,22 @@ public void scrollToBottom() {
...
+      PageObjectLogging.log("Offset", "Offset is " + String.valueOf(offset), true);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java]: MINOR Directly append the argument of String.valueOf(). [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1153)

File: src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java [hunk 3]
```diff
@@ -136,9 +135,9 @@ public void scrollToElementInModal(WebElement element, WebElement modal) {
   }
 
   public void scrollElementIntoViewPort(WebElement element) {
-    if (!isElementInViewPort(element)) {
+    //if (!isElementInViewPort(element)) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java [hunk 4]
```diff
@@ -136,9 +135,9 @@ public void scrollToElementInModal(WebElement element, WebElement modal) {
...
       scrollToElement(element);
-    }
+    //}
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [JavascriptActions.java#L74](https://github.com/Wikia/selenium-tests/blob/69539f14977048d28b70b028089485c3628e9c5b/src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java#L74): Remove this unused private "isElementInViewPort" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [JavascriptActions.java#L189](https://github.com/Wikia/selenium-tests/blob/69539f14977048d28b70b028089485c3628e9c5b/src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java#L189): Immediately return this expression instead of assigning it to the temporary variable "value". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1488)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [JavascriptActions.java#L74](https://github.com/Wikia/selenium-tests/blob/9394fd2f6c77dc02cd81f852cb3166c1fd114748/src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java#L74): Remove this unused private "isElementInViewPort" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [JavascriptActions.java#L189](https://github.com/Wikia/selenium-tests/blob/9394fd2f6c77dc02cd81f852cb3166c1fd114748/src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java#L189): Immediately return this expression instead of assigning it to the temporary variable "value". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1488)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [JavascriptActions.java#L188](https://github.com/Wikia/selenium-tests/blob/92fbd684dff6d5d03a585681a3cc6c39fc27fdab/src/test/java/com/wikia/webdriver/common/core/elemnt/JavascriptActions.java#L188): Immediately return this expression instead of assigning it to the temporary variable "value". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1488)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sczerwinski-wikia** [comment]
To check how it works run userCanWriteMessageInSourceMode with User.HELPER on staging

**Pull request merged**